drivers/include/soft_spi.h

```diff
@@ -0,0 +1,286 @@
+ *           the second rising SCK edge.
+ * - MODE_2: CPOL=1, CPHA=0 - The first data bit is sampled by the receiver on
+ *           the first falling SCK edge.
+ * - MODE_3: CPOL=1, CPHA=1 - The first data bit is sampled by the receiver on
+ *           the second falling SCK edge.
+ */
+#ifndef HAVE_SOFT_SPI_MODE_T
+typedef enum {
+    SOFT_SPI_MODE_0 = 0,         /**< CPOL=0, CPHA=0 */
+    SOFT_SPI_MODE_1,             /**< CPOL=0, CPHA=1 */
+    SOFT_SPI_MODE_2,             /**< CPOL=1, CPHA=0 */
+    SOFT_SPI_MODE_3              /**< CPOL=1, CPHA=1 */
+} soft_spi_mode_t;
+#endif
+
+/**
+ * @brief   Available SPI clock speeds
+ *
+ * The actual speed of the bus can vary to some extend, as the combination of
+ * CPU clock and available prescaler values on certain platforms may not make
+ * the exact values possible.
+ */
+#ifndef HAVE_SOFT_SPI_CLK_T
+typedef enum {
+    SOFT_SPI_CLK_100KHZ = 0,     /**< drive the SPI bus with 100KHz */
+    SOFT_SPI_CLK_400KHZ,         /**< drive the SPI bus with 400KHz */
+    SOFT_SPI_CLK_1MHZ,           /**< drive the SPI bus with 1MHz */
+    SOFT_SPI_CLK_5MHZ,           /**< drive the SPI bus with 5MHz */
+    SOFT_SPI_CLK_10MHZ           /**< drive the SPI bus with 10MHz */
+} soft_spi_clk_t;
+#endif
+
+/**
+ * @brief Software SPI port descriptor
+ */
+typedef struct {
+    gpio_t miso_pin;              /**< MOSI pin */
+    gpio_t mosi_pin;              /**< MOSI pin */
+    gpio_t clk_pin;               /**< CLK pin */
+    soft_spi_mode_t soft_spi_mode;          /**< data and clock polarity */
+} soft_spi_conf_t;
+
+/**
+ * @brief   Basic initialization of the given SPI bus
+ *
+ * This function does the basic initialization including pin configuration for
+ * MISO, MOSI, and CLK pins. After initialization, the given device should be
+ * in power down state.
+ *
+ * This function is intended to be called by the board initialization code
+ * during system startup to prepare the (shared) SPI device for further usage.
+ * It uses the board specific initialization parameters as defined in the
+ * board's `periph_conf.h`.
+ *
+ * Errors (e.g. invalid @p bus parameter) are not signaled through a return
+ * value, but should be signaled using the assert() function internally.
+ *
+ * @note    This function MUST not be called more than once per bus!
+ *
+ * @param[in] bus       SPI device to initialize
+ */
+void soft_spi_init(soft_spi_t bus);
+
+/**
+ * @brief   Initialize the used SPI bus pins, i.e. MISO, MOSI, and CLK
+ *
+ *
+ * After calling soft_spi_init, the pins must be initialized (i.e. soft_spi_init is
+ * calling this function internally). In normal cases, this function will not be
+ * used. But there are some devices (e.g. CC110x), that use SPI bus lines also
+ * for other purposes and need the option to dynamically re-configure one or
+ * more of the used pins. So they can take control over certain pins and return
+ * control back to the SPI driver using this function.
```

Comment:
IMO it is not needed to keep the whole copy-pasta text here. Just state what the function does and that pins should be initialized after `spi_init`

Reply:
copy paste error again
:heavy_check_mark: